aikolesnikov/src/main/java/library/Library.java

```diff
@@ -0,0 +1,210 @@
+package library;
+
+import datastructures.MyArrayList;
+
+import java.util.Comparator;
+
+import static library.LIB_SETTINGS.BOOK_RENT_LIMIT;
+import static library.LIB_SETTINGS.LIBRARY_NAME;
+
+/**
+ * Library
+ */
+public class Library {
+
+    private String name;
+    private MyArrayList books;
+    private MyArrayList clients;
+
+    Library() {
+        name = LIBRARY_NAME;
+        books = new MyArrayList();
+        clients = new MyArrayList();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public MyArrayList getBooks() {
+        return books;
+    }
+
+    public MyArrayList getClients() {
+        return clients;
+    }
+
+    //  1) посмотреть список читателей
+    public String printListOfClients() {
+        final StringBuilder sb = new StringBuilder("");
+        for (int i = 0; i < clients.size(); i++) {
+            sb.append(clients.get(i)).toString();
+            sb.append("};\n");
+        }
+        return sb.toString();
+    }
+
+    //  2) посмотреть список доступных конкретных изданий
+    public String printListOfBooks() {
+        final StringBuilder sb = new StringBuilder("");
+        for (int i = 0; i < books.size(); i++) {
+            sb.append(books.get(i)).toString();
+            sb.append("};\n");
+        }
+        return sb.toString();
+    }
+
+    // 3) добавить издание в библиотеку.
+    boolean addBook(BookCopy book) {
+        return (book == null) || books.contains(book) ? false : books.add(book);
+    }
+
+    //  4) добавить читателя в список читателей
+    boolean addReader(Client client) {
+        return (client == null) || clients.contains(client) ? false : clients.add(client);
+    }
+
+    //  5) выдать издание читателю (если оно есть в наличии).
+    // Читателю запрещается брать больше 3-х изданий.
+    boolean getBook(BookCopy book, Client client) {
+        if ((book == null) || !(books.contains(book))) {
+            System.out.println("Book does not exists.");
+            return false;
+        }
+        if ((client == null) || !(clients.contains(client))) {
+            System.out.println("Client does not exist ");
+            return false;
+        }
+        if (client.isBlocked()) {
+            System.out.println("Client is not allowed to rent books.");
+            return false;
+        }
+
+        if (client.getReadBooks().size() == BOOK_RENT_LIMIT) {
+            System.out.println("Client's rent limit is exceeded.");
+            return false;
+        }
+
+        if (books.remove(book))
+            client.getReadBooks().add(book);
+
+        return true;
+    }
+
+    boolean returnBook(Client client, BookCopy book) {
+        if ((book == null) || !(client.getReadBooks().contains(book))) {
+            System.out.println("Book does not exists.");
+            return false;
+        }
+        if ((client == null) || !(clients.contains(client))) {
```

Comment:
@Test
    public void checkGetReturnBook(){
        // init
        Client client = new Client();
        lib.addReader(client);
        BookCopy book = new BookCopy();;
        book.setName("Book1");
        lib.addBook(book);
        assertEquals(lib.getBooks().size(), 1);
        assertEquals(client.getReadBooks().size(), 0);

        // good book to wrong client )
        lib.getBook(book, null);
        assertEquals(lib.getBooks().size(), 1);
        assertEquals(client.getReadBooks().size(), 0);
        // wrong book to good client )
        lib.getBook(null, client);
        assertEquals(lib.getBooks().size(), 1);
        assertEquals(client.getReadBooks().size(), 0);
        // good book to good client )
        lib.getBook(book, client);
        assertEquals(lib.getBooks().size(), 0);
        assertEquals(client.getReadBooks().size(), 1);

        // good book from wrong client )
        lib.returnBook(null, (BookCopy) client.getReadBooks().get(0));
        assertEquals(lib.getBooks().size(), 0);
        assertEquals(client.getReadBooks().size(), 1);
        // wrong book from good client )
        lib.returnBook(client, null);
        assertEquals(lib.getBooks().size(), 0);
        assertEquals(client.getReadBooks().size(), 1);
        // good book from good client )
        lib.returnBook(client, (BookCopy) client.getReadBooks().get(0));
        assertEquals(lib.getBooks().size(), 1);
        assertEquals(client.getReadBooks().size(), 0);
    }